src/frontend/src/optimizer/plan_node/logical_over_agg.rs

```diff
@@ -153,75 +162,115 @@ impl LogicalOverAgg {
                         4847.into(),
                     )
                     .into());
                 }
             }
         }
-        if window_funcs.len() > 1 {
-            return Err(ErrorCode::NotImplemented(
-                "Multiple window functions".to_string(),
-                None.into(),
-            )
-            .into());
-        }
-        let WindowFunction {
-            args,
-            return_type,
-            function_type,
-            partition_by,
-            order_by,
-        } = window_funcs.into_iter().next().unwrap();
-        assert!(args.is_empty());
-        assert!(return_type == DataType::Int64);
 
-        // TODO: rewrite ORDER BY & PARTITION BY expr to InputRef like `LogicalAgg`
-        let order_by = order_by
+        let plan_window_funcs = window_funcs
+            .into_iter()
+            .map(Self::convert_window_function)
+            .try_collect()?;
+
+        let over_agg = Self::new(plan_window_funcs, input);
+        Ok((over_agg.into(), select_exprs))
+    }
+
+    fn convert_window_function(window_function: WindowFunction) -> Result<PlanWindowFunction> {
+        // TODO: rewrite expressions in `ORDER BY`, `PARTITION BY` and arguments to `InputRef` like
+        // in `LogicalAgg`
+        let order_by: Vec<_> = window_function
+            .order_by
             .sort_exprs
             .into_iter()
             .map(|e| match e.expr.as_input_ref() {
                 Some(i) => Ok(ColumnOrder::new(i.index(), e.order_type)),
                 None => Err(ErrorCode::NotImplemented(
                     "ORDER BY expression in window function".to_string(),
                     None.into(),
-                )
-                .into()),
+                )),
             })
-            .collect::<Result<Vec<_>>>()?;
-        let partition_by = partition_by
+            .try_collect()?;
+        let partition_by: Vec<_> = window_function
+            .partition_by
             .into_iter()
             .map(|e| match e.as_input_ref() {
                 Some(i) => Ok(*i.clone()),
                 None => Err(ErrorCode::NotImplemented(
                     "PARTITION BY expression in window function".to_string(),
                     None.into(),
-                )
-                .into()),
+                )),
             })
-            .collect::<Result<Vec<_>>>()?;
-
-        let over_agg = Self::new(
-            PlanWindowFunction {
-                function_type,
-                return_type,
-                partition_by,
-                order_by,
-            },
-            input,
-        );
-        Ok((over_agg.into(), select_exprs))
+            .try_collect()?;
+
+        let mut args = window_function.args;
+        let frame = match window_function.kind {
+            WindowFuncKind::RowNumber | WindowFuncKind::Rank | WindowFuncKind::DenseRank => {
+                // ignore frame for rank functions
+                None
```

Comment:
LogicalWindow shouldn't override the original SQL. This limitation is all about `StreamWindow`, so we can detect it in the `StreamWindow` and then throw an error.

Reply:
Seems the frame is ignored for rank functions in PG.